memory/ndctl.py

```diff
@@ -0,0 +1,277 @@
+                vals.append(self.get_json_val(nid, self.opt_dict[option]))
+            return vals
+        return json_op
+
+    @staticmethod
+    def get_json_val(json_op, field):
+        """
+        Get the value of a field in given json
+        """
+        for key, value in json_op.iteritems():
+            if key == field:
+                return value
+        return None
+
+    @staticmethod
+    def check_buses():
+        """
+        Get buses from sys subsystem to verify persisment devices exist
+        """
+        return glob.glob('/sys/bus/nd/drivers/nd_bus/ndbus*')
+
+    def disable_region(self, name='all'):
+        """
+        Disable given region
+        """
+        if process.system('%s disable-region %s' % (self.binary, name),
+                          shell=True, ignore_status=True):
+            self.fail("Failed to disable %s region(s)" % name)
+
+    def enable_region(self, name='all'):
+        """
+        Enable given region
+        """
+        if process.system('%s enable-region %s' % (self.binary, name),
+                          shell=True, ignore_status=True):
+            self.fail("Failed to enable %s region(s)" % name)
+
+    def disable_namespace(self, namespace='all', region='', bus='',
+                          force=False):
+        """
+        Disable namepsaces
+        """
+        args = namespace
+        if region:
+            args = '%s -r %s' % (args, region)
+        if bus:
+            args = '%s -b %s' % (args, bus)
+        if force:
+            args = '%s -f' % args
+
+        if process.system('%s disable-namespace %s' % (self.binary, args),
+                          shell=True, ignore_status=True):
+            self.fail('Namespace disble command failed')
+
+    def create_namespace(self, region='', bus='', n_type='pmem', mode='fsdax',
+                         memmap='mem', name='', size='', uuid='',
+                         sector_size='', align='', force=False,
+                         autolabel=False):
+        """
+        Creates namespace with specified options
+        """
+        args_dict = {region: '-r', bus: '-b', name: '-n', size: '-s', uuid: '-u', sector_size: '-l', align: '-a'}
+        args = '-t %s -m %s ' % (n_type, mode)
+
+        for option in list(args_dict.keys()):
+            if option:
+                args += ' %s %s' % (args_dict[option], option)
+        if mode in ['fsdax', 'devdax']:
+            args += ' -M %s'% memmap
+        if force:
+            args += ' -f'
+        if autolabel:
+            args += ' -L'
+
+        if process.system('%s create-namespace %s' % (self.binary, args),
+                          shell=True, ignore_status=True):
+            self.fail('Namespace create command failed')
+
+    def destroy_namespace(self, namespace='all', region='', bus='',
+                          force=False):
+        """
+        Destroy namepsaces
+        """
+        args = namespace
+        if region:
```

Comment:
could be similar to https://github.com/avocado-framework-tests/avocado-misc-tests/pull/1144/files#diff-bb3bf4b7fe06bf39ad148236e4b11a3dR116

Reply:
in fact, these both can use the same dict

Reply:
ok

Reply:
I meant initiating them at class scope.